public/app/features/scenes/dashboard/PanelMenuBehavior.tsx

```diff
@@ -0,0 +1,37 @@
+import { locationUtil, PanelMenuItem } from '@grafana/data';
+import { locationService } from '@grafana/runtime';
+import { VizPanel, VizPanelMenu } from '@grafana/scenes';
+import { t } from 'app/core/internationalization';
+
+/**
+ * Behavior is called when VizPanelMenu is activated (ie when it's opened).
+ */
+export function panelMenuBehavior(menu: VizPanelMenu) {
+  // hm.. add another generic param to SceneObject to specify parent type?
```

Comment:
And default to SceneObject? I mean, this generic would only help us in scenarios like this. Don't think it would be supportive when implementing a scene. But - this could actually be helpful for validating objects like this https://github.com/grafana/scenes/blob/main/packages/scenes/src/components/layout/SceneFlexLayout.tsx#L86. Not sure tho how the validation would work, we cannot use `instanceof T` in TS.

Reply:
Yea, as thinking of the Layouts that could type the paren. But maybe it's too much of a niche to warrant that second generic param